rpc/swap.go

```diff
@@ -174,8 +235,9 @@ type GetStageRequest struct {
 
 // GetStageResponse ...
 type GetStageResponse struct {
-	Stage string `json:"stage"`
-	Info  string `json:"info"`
+	Stage     string    `json:"stage"`
```

Comment:
Can we update the type to `types.Status`? I incorrectly assumed this was the contract stage until I looked down at where it is filled in.

We really need some renames if it is not too far out of scope for this PR. Based on the name, I would expect the next field to come from `swap.Info`, but it is actually a textual `description` of the Status.

Reply:
If we need `MarshalText` / `UnmarshalText` methods for the Status type, I can add them to this PR.

Reply:
updated. I'd rather do other changes in another PR since I'd like to get this in asap as I've been using it for testing

Reply:
I added the Status marshalling support here:
https://github.com/AthanorLabs/atomic-swap/pull/323

I tried to do it in a way that would create minimal merge conflicts.